src/Symfony/Component/Validator/Constraints/EnumValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * EnumValidator validates that the value is one of the expected values.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class EnumValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Enum) {
+            throw new UnexpectedTypeException($constraint, Enum::class);
+        }
+
+        if (!enum_exists($constraint->type) || !method_exists($constraint->type, 'from')) {
+            throw new ConstraintDefinitionException('The "type" must be an \BackedEnum.');
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        try {
+            $value = $constraint->type::from($value);
+        } catch (\ValueError) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setParameter('{{ choices }}', $this->formatValues($constraint->type::cases()))
+                ->setCode(Choice::NO_SUCH_CHOICE_ERROR)
+                ->addViolation();
+        }
+
+        if (\count($constraint->only) > 0 && !\in_array($value, $constraint->only, true)) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setParameter('{{ choices }}', $this->formatValues($constraint->only))
+                ->setCode(Choice::NO_SUCH_CHOICE_ERROR)
+                ->addViolation();
+        }
+
+        if (\count($constraint->except) > 0 && \in_array($value, $constraint->except, true)) {
+            $reflector = new \ReflectionClass($constraint->type);
+            $constants = $reflector->getConstants();
+
+            foreach ($constants as $name => $value) {
+                if (\in_array($value, $constraint->except, strict: true)) {
+                    unset($constants[$name]);
+                }
+            }
```

Comment:
https://www.php.net/array_diff

Reply:
You've ignored this comment. 🙃

Reply:
I have tried but didn't find how you wanted to use it for the moment. This code does not work that's why I didn't resolve this conversation

Example : 

```php
enum MyEnum: string
{
    case YES = 'yes';
    case NO = 'no';
}

array_diff(MyEnum::cases(), [MyEnum::YES]);
```

Reply:
Ah yes, I forgot that `array_diff()` requires the array values to be stringable. 🤦🏻‍♂️ 

`array_udiff()` with `static fn ($a, $b) => $a <=> $b` as comparison function should work though.

Reply:
Didn't find better than this for the moment : 

```php
$values = array_filter($values, function ($currentValue) use ($constraint) {
    return !\in_array($currentValue, $constraint->except, true);
});
```